OpenSim/Common/Component.cpp

```diff
@@ -615,86 +615,27 @@ void Component::setParent(const Component& parent)
 
 std::string Component::getFullPathName() const
```

Comment:
Ah, for example, `getRelativePathName()` must convert the resulting string back into a `ComponentPath`, even though `getFullPathName()` had created a `ComponentPath` initially. So perhaps such a variant could be useful?